pkg/dns/dns.go

```diff
@@ -245,106 +244,111 @@ maesh:53 {
+	if err := c.patchKubeDNSConfig(kubeDNSDeployment, coreDNSServiceIP); err != nil {
 		return err
 	}
 
-	c.logger.Debug("Patching CoreDNS configmap")
-
 	if err := c.ConfigureCoreDNS(maeshNamespace, clusterDomain, maeshNamespace); err != nil {
 		return err
 	}
 
-	if err := c.restartPods(deployment); err != nil {
+	if err := c.restartPods(kubeDNSDeployment); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// getKubeDNSConfigMap parses the deployment and returns the associated configuration configmap.
-func (c *Client) getKubeDNSConfigMap(kubeDeployment *appsv1.Deployment) (*corev1.ConfigMap, error) {
-	for _, volume := range kubeDeployment.Spec.Template.Spec.Volumes {
-		if volume.ConfigMap == nil {
-			continue
-		}
+func (c *Client) patchKubeDNSConfig(deployment *appsv1.Deployment, coreDNSServiceIP string) error {
+	configMap, err := c.getOrCreateKubeDNSConfigMap(deployment)
+	if err != nil {
+		return err
+	}
 
-		cfgMap, err := c.kubeClient.CoreV1().ConfigMaps(kubeDeployment.Namespace).Get(volume.ConfigMap.Name, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
-		}
+	stubDomains := make(map[string][]string)
 
-		return cfgMap, nil
+	if stubDomainsStr := configMap.Data["stubDomains"]; stubDomainsStr != "" {
+		if err = json.Unmarshal([]byte(stubDomainsStr), &stubDomains); err != nil {
+			return fmt.Errorf("unable to unmarshal stub domains: %w", err)
+		}
 	}
 
-	return nil, errors.New("corefile configmap not found")
-}
+	stubDomains["maesh"] = []string{coreDNSServiceIP}
 
-func (c *Client) patchKubeDNSConfigMap(kubeConfigMap *corev1.ConfigMap, namespace, coreDNSIp string) error {
-	originalBlock, exist := kubeConfigMap.Data["stubDomains"]
-	if !exist {
-		originalBlock = "{}"
+	configMapData, err := json.Marshal(stubDomains)
+	if err != nil {
+		return fmt.Errorf("unable to marshal stub domains: %w", err)
 	}
 
-	stubDomains := make(map[string][]string)
-	if err := json.Unmarshal([]byte(originalBlock), &stubDomains); err != nil {
+	configMap.Data["stubDomains"] = string(configMapData)
+
+	if _, err := c.kubeClient.CoreV1().ConfigMaps(configMap.Namespace).Update(configMap); err != nil {
 		return err
 	}
 
-	stubDomains["maesh"] = []string{coreDNSIp}
-
-	var newData []byte
+	return nil
+}
 
-	newData, err := json.Marshal(stubDomains)
-	if err != nil {
-		return err
-	}
+// getOrCreateKubeDNSConfigMap parses the deployment and returns the kube-dns ConfigMap. If the associated ConfigMap
+// volume is marked as optional and the ConfigMap is not found, this method will create and return the created ConfigMap.
+func (c *Client) getOrCreateKubeDNSConfigMap(deployment *appsv1.Deployment) (*corev1.ConfigMap, error) {
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		if volume.ConfigMap == nil {
+			continue
+		}
 
-	if kubeConfigMap.Data == nil {
-		kubeConfigMap.Data = make(map[string]string)
-	}
+		if volume.ConfigMap.Name != "kube-dns" {
```

Comment:
I thought exactly the same, and I don't have a strong opinion. My thinking is that in Kubernetes documentations they always talk about a `kubedns` configmap, so that's why I've done it.

You're right, I think this should be configurable.